ui-components/src/main/java/org/dhis2/ui/dialogs/bottomsheet/BottomSheetDialog.kt

```diff
@@ -33,13 +33,14 @@ import org.hisp.dhis.mobile.ui.designsystem.component.Button
 import org.hisp.dhis.mobile.ui.designsystem.component.ButtonBlock
 import org.hisp.dhis.mobile.ui.designsystem.component.ButtonStyle
 import org.hisp.dhis.mobile.ui.designsystem.theme.Border
+import org.hisp.dhis.mobile.ui.designsystem.theme.DHIS2Theme
 import org.hisp.dhis.mobile.ui.designsystem.theme.Spacing.Spacing24
 import org.hisp.dhis.mobile.ui.designsystem.theme.SurfaceColor
 import org.hisp.dhis.mobile.ui.designsystem.theme.TextColor
 
 class BottomSheetDialog(
     var bottomSheetDialogUiModel: BottomSheetDialogUiModel,
-    var onMainButtonClicked: () -> Unit = {},
+    var onMainButtonClicked: ((org.dhis2.ui.dialogs.bottomsheet.BottomSheetDialog)) -> Unit = {},
```

Comment:
just a comment of what I like to do, because a don't like having to declare all that long name to specify a class. You could use kotlin's import aliases. So you could do something like the following. 

``` kotlin
import org.dhis2.ui.dialogs.bottomsheet.BottomSheetDialog as MyCustomDialog

...
var onMainButtonClicked: (MyCustomDialog) -> Unit = {}

```

It's requirement to change it, just so that you know we have that option.

Reply:
Thanks for the comment Ferdy! I will keep that in mind for next time!